### ISSUE TITLE
Profile input/output/connector name lookups; create indices if necessary.

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#371]
While Xuefeng was measuring my sarcomere lengths, I was wondering if input/output/connector name lookups will be expensive. If we do profiling and determine it'd be nice to access inputs/outputs/connectors by index instead of by string name, then we can allow access by index. The actual data structure could be a vector, and there can be a map<string, int> that allows name access.

This might be a moot point since the way these 3 are defiined will change when vs gets its act together.

BUT, there might be components that need N inputs (e.g., a summer). Consider this.

**Comment by aseth1** [comment]
I agree. String lookups are intended for convenience and to ensure proper structure. We should always have either an equivalent index accessor or be able to hang on to a handle of the thing we are looking up by name.

@tgeijten  ran into this issue with muscle accessing the same cache entry repeatedly by name. The good thing is that these will be relatively easy to fix.
